### ISSUE TITLE
Fix double focus styles on certain inputs

### ISSUE THREAD

**Issue by orion-cengage** [cengage/react-magma#799]
Perkins CENG_2485, CENG_2494, CENG_2495

This issue impacts password, date input, and search input.

When focus moves to the button on the right side of the input, the focus styling on the overall input should not be visible.  Essentially, you shouldn't have two things "focused" at the same time.

![image](https://user-images.githubusercontent.com/24397282/175553695-d2b52450-8834-49a3-9524-ce83ed7fa3df.png)

![image](https://user-images.githubusercontent.com/24397282/175553743-5078fc4d-5814-4d46-b093-c936106a4788.png)

![image](https://user-images.githubusercontent.com/24397282/175553810-90e5c9ce-298c-4011-9410-ce333bc68a12.png)

**Testing Notes:**
- Search:
-- `size`: `medium` and `large`
-- `isClearable`: `true` and `false`

- Input
-- `icon`" with and without one
-- `isClearable`: `true` and `false`
-- `isClearable`: with and without an Icon
-- `onIconClick`: with a function being passed and without
-- `iconPosition`: `left`, `top`, `right`

- DatePicker

**Comment by Uditsingh67** [comment]
Regressed the ticket , Below are the observations:
**See Screen-cast :** http://somup.com/c3jDnku8xw

**Testing Notes:**
Search: 
-- size: medium and large : **PASS**
-- isClearable: true and false : **PASS**
**See Screen-cast :** http://somup.com/c3jDnmu8xH

Input
-- icon" with and without one : **PASS**
-- isClearable: true and false : **PASS**
-- isClearable: with and without an Icon : **PASS**
-- onIconClick: with a function being passed and without : **FAIL**
-- iconPosition: left, top, right : **FAIL ( Icon Position is not changed for Clickable icon &  Clickable icon Large components )**
**See Screen-cast :** http://somup.com/c3jDn8u8yQ

DatePicker : **PASS**

**Comment by silvalaura** [comment]
@Uditsingh67 This issue is not concerned with the placement of the icons, it's concerned with the focus on inputs. We care if the input highlighting is causing any issues, but it appears that the issues called out here are unrelated. 

**-- onIconClick: with a function being passed and without**
onIconClick with a function can be tested with the [Clickable Icon story](https://storybook-preview-dev--upbeat-sinoussi-f675aa.netlify.app/?path=/story/input--clickable-icon). without a function can be tested with [Icon Right](https://storybook-preview-dev--upbeat-sinoussi-f675aa.netlify.app/?path=/story/input--icon-right). 

The test is marked as fail with no explanation.

**-- iconPosition: left, top, right** 
This issue with the icon not moving is unrelated to this issue. It can be tested on these stories.
[Icon Left](https://storybook-preview-dev--upbeat-sinoussi-f675aa.netlify.app/?path=/story/input--icon-left)
[Icon Top](https://storybook-preview-dev--upbeat-sinoussi-f675aa.netlify.app/?path=/story/input--icon-top)
[Icon Right](https://storybook-preview-dev--upbeat-sinoussi-f675aa.netlify.app/?path=/story/input--icon-right)

**Comment by Uditsingh67** [comment]
The issue is fixed.

<html>
<body>


OS | Version | Result
-- | -- | --
Win 10 (Chrome) | 102.0.5005.63 | PASS (http://somup.com/c3jbiYu9Yo)
Win 10 (Firefox) | 101.0 | PASS (http://somup.com/c3jbi0u9Yw)
Win 10 (Edge) | 103.0.1264.37 | PASS (http://somup.com/c3jbibu9Yn)
MAC (safari) | 15.3 | Pass (http://somup.com/c3jb1ou9oL)
Ipad (safari) | v15.1 | Pass (http://somup.com/c3jb1uu9DC)


</body>
</html>

**Comment by Uditsingh67** [comment]
Issue does not exist.
See Screen-cast : http://somup.com/c3jwfNU2bv
